resources/iam-user-mfa-attachment.go

```diff
@@ -0,0 +1,86 @@
+package resources
+
+import (
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/rebuy-de/aws-nuke/pkg/types"
+)
+
+type IAMUserMfaAttachment struct {
+	svc          *iam.IAM
+	userName     *string
+	serialNumber string
+}
+
+func init() {
+	register("IAMUserMfaAttachment", ListIAMUserMfaAttachments)
+}
+
+func ListIAMUserMfaAttachments(sess *session.Session) ([]Resource, error) {
+	results, err := ListIAMUsers(sess)
+	if err != nil {
+		return nil, err
+	}
+
+	resources := make([]Resource, 0)
+	for _, resource := range results {
+		user := resource.(*IAMUser)
+		userMfaDevices, err := user.ListAttachedMFADevices()
+		if err != nil {
+			return nil, err
+		}
+		resources = append(resources, userMfaDevices...)
+	}
+
+	return resources, nil
+}
+
+func (f *IAMUser) ListAttachedMFADevices() ([]Resource, error) {
```

Comment:
Is there no way to list all devices? This is in the end n+1 thingy here, and we are calling the API for every user once. Could slow down a lot.

Reply:
That's true. I think the `ListMFADevice` just does not need to have a `UserName` argument. At least [the docs](https://docs.aws.amazon.com/sdk-for-go/api/service/iam/#ListMFADevicesInput) do not mention it and the equivalent CLI command `aws iam list-mfa-devices` does not need it.

Reply:
and I guess the same thing goes for ssh keys ( new thing in IAM that also prevents removing an user completely).
I just saw other comments and I will work on the proposed changes.